app/db/interface.py

```diff
@@ -134,7 +127,10 @@ def get_payments(self, payment_ids=None, payment_attributes=None):
         else:
             payments = self.db.session.query(Payment).all()
 
-        payments = self._remove_soft_deletes(payments)
+        if len(payments) == 0:
```

Comment:
Just a neat Python thing: you can actually rewrite this as `if payments` and that will amount to the same thing.